DBFileReaderLib/Readers/HTFXReader.cs

```diff
@@ -185,9 +185,38 @@ public HTFXReader(Stream stream)
                     reader.BaseStream.Position += 32; // sha hash
                 }
 
+                long length = reader.BaseStream.Length;
+
+                // Version 8 was first seen in 9.1.0.39291 with no actual format changes, likely by error. A new field was added later in build 2.5.2.39570.
+                // This means we have to 'detect' which of the formats this actually is...
+                // NOTE: This method will likely fail on files that don't have at least 2 hotfix entries in them.
+                if (Version == 8 && length > (reader.BaseStream.Position + 24))
+                {
+                    // Save position to go back to later
+                    var prePos = reader.BaseStream.Position;
+
+                    if (reader.ReadUInt32() != HTFXFmtSig)
+                        throw new Exception("Invalid hotfix entry magic!");
+
+                    reader.BaseStream.Position += 16; // Skip ahead by 16 bytes, this includes the 4 bytes _actually_ added in V8
+
+                    var dataSize = reader.ReadInt32();
+
+                    reader.BaseStream.Position += 4 + dataSize; // Skip ahead by 4 bytes + size of hotfix data
+
+                    // Check if we encounter a hotfix signature in the next hotfix record
+                    if (reader.ReadUInt32() != HTFXFmtSig)
```

Comment:
Maybe check First if there is still data left to read.
Because if there is only 1 hotfix there is probably no data left to read.

Reply:
There's another bounds issue here that I also think I resolved (datasize could not be datasize if the header is the wrong version) in the new commit, it's a bit of a hack either way that I kind of want to get rid of entirely but not sure how yet. :(